gtda/sampling/sample_point_clouds.py

```diff
@@ -0,0 +1,177 @@
+"""Point-cloud–related plotting functions and classes."""
+# License: GNU AGPLv3
+
+import numpy as np
+from scipy.optimize import root_scalar
+
+from gtda.utils import validate_params
+from gtda.utils.intervals import Interval
+
+
+def sphere_sampling(n=1000, r=1, noise=0):
+    """Uniformly and randomly samples points from a round
+    sphere centered at the origin of 3-space.
+    
+    Parameters
+    ----------
+    n : int or None, optional, default: 1000
+        The number of points to be sampled.
+
+    r : float or None, optional, default: 1
+        The radius of the sphere to be sampled from.
+        Must be a positive number.
+
+    noise : float or None, optional, default: 0
+        The noise of the sampling, which is introduced by
+        adding Gaussian noise around each data point.
+        Must be a non-negative number.
+
+    Returns
+    -------
+    points : ndarray of shape (n, 3).
+        NumPy-array containing the sampled points.
+    """
+
+    validate_params({"n": n},
+                    {"n": {"type": (int,),
+                           "in": Interval(0, np.inf, closed="neither")}})
+    validate_params({"r": r},
+                    {"r": {"type": (int, float),
+                           "in": Interval(0, np.inf, closed="neither")}})
+    validate_params({"noise": noise},
+                    {"noise": {"type": (int, float),
+                               "in": Interval(0, np.inf, closed="left")}})
+
+    def parametrization(theta, phi):
+        x = r*np.cos(theta)*np.sin(phi)
+        y = r*np.sin(theta)*np.sin(phi)
+        z = r*np.cos(phi)
+        return np.array([x, y, z])
+    U_phi_inverse = (lambda y: np.arccos(1-2*y))
+    points = np.zeros(shape=(n, 3))
+    for i in range(n):
+        theta = np.random.uniform(low=0, high=2*np.pi)
+        phi = U_phi_inverse(np.random.uniform())
+        pt = parametrization(theta, phi)
```

Comment:
you are using the method parametrisation just here in this for loop.

Just write
```python
pt = np.array(
    [
        r*np.cos(theta)*np.sin(phi),
        r*np.sin(theta)*np.sin(phi),
        r*np.cos(phi),
    ]
)
```

it is much easier to understand what it is happening

And use another name for the variable pt. Is it a point? Just use the name point.